### PR TITLE
Token revamp

### PR DESCRIPTION
**Pull request by nfegard**

### CODE REVIEW

File: token/token_test.go [hunk 1]
```diff
@@ -1,38 +1,172 @@
...
-func (a A) GetCurrentUnixTimestamp() string {
-	return strconv.FormatInt(a.Now().Unix(), 10)
+func TestTokenIsSignedWithKeyWrongSignature(t *testing.T) {
+	incorrectSignature, err := base64.StdEncoding.DecodeString("c2ln")
+	assert.Ok(t, err)
+	testToken := Token{UserIdentifier: "steve", Timestamp: getEpochTime(), Signature: incorrectSignature}
+	actual, err := testToken.IsSignedWithKey([]byte(testKey))
+
+	assert.Ok(t, err)
+	assert.Assert(t, !actual, "The token is failing to be recognized as having an incorrect signature")
```
Comment (stevex86) [inline token/token_test.go]: I think this error condition is better described as: "Erroneously confirmed that an invalid signature was signed with a valid key."

File: token/token_test.go [hunk 2]
```diff
@@ -1,38 +1,172 @@
...
 
-func TestMakeTokenHappyPath(t *testing.T) {
-	_, err := MakeToken("steve", []byte(key))
+func TestTokenIsSignedWithKeyWrongKey(t *testing.T) {
+	testSignature := generateTestSignature()
+	testToken := Token{UserIdentifier: "steve", Timestamp: getEpochTime(), Signature: testSignature}
+	actual, err := testToken.IsSignedWithKey(nil)
+
 	assert.Ok(t, err)
+	assert.Assert(t, !actual, "The token is failing to be recognized as having an incorrect signature")
```
Comment (stevex86) [inline token/token_test.go]: I think this error condition is better described as: "Erroneously confirmed that an valid signature was signed with a invalid key."

**Comment by stevex86** [comment]
Would we rather merge this branch straight back to `develop`? Or to `token-err-test-improvements` and then back to `develop`?

**Comment by nfegard** [comment]
Yeah, doesn't it make more sense that way?

**Comment by tforrest** [comment]
+1

**Pull request merged**